client.go

```diff
@@ -84,6 +85,13 @@ func BearerAuth(token string) func(*Request) {
 	}
 }
 
+func BasicAuth(username, password string) RequestParam {
+	return func(req *Request) {
+		auth := base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
+		req.Header.Set("Authorization", "Basic "+auth)
```

Comment:
Besser `req.SetBasicAuth(...)` dann braucht man sich nicht um die Details zu kümmern.

Reply:
du meinst wahrscheinlich: req.RawRequest.SetBasicAuth(..)

Reply:
Das geht nicht da der RawRequest erst erstellt wird nachdem die RequestParams angewendet wurden. In dem Moment wo diese Funktion ausgeführt wird ist der RawRequest somit noch nil. Auf den ersten Blick würde aber nichts dagegen sprechen diese Reihenfolge zu änderen

Reply:
Bei genauerer Betrachtung leider schon. Also das zu ändern wäre möglich aber etwas aufwändiger als nur die Zeilen tauschen